feature-account-impl/src/main/java/io/novafoundation/nova/feature_account_impl/data/proxy/RealProxySyncService.kt

```diff
@@ -0,0 +1,166 @@
+import io.novafoundation.nova.feature_account_api.domain.model.LightMetaAccount
+import io.novafoundation.nova.feature_account_api.domain.model.MetaAccount
+import io.novafoundation.nova.feature_account_api.domain.model.MetaAccountId
+import io.novafoundation.nova.feature_account_api.domain.model.accountIdIn
+import io.novafoundation.nova.runtime.ext.addressOf
+import io.novafoundation.nova.runtime.ext.isSubstrateBased
+import io.novafoundation.nova.runtime.multiNetwork.ChainRegistry
+import io.novafoundation.nova.runtime.multiNetwork.chain.model.Chain
+import io.novafoundation.nova.runtime.multiNetwork.chain.model.ChainId
+import io.novafoundation.nova.runtime.multiNetwork.findChains
+import jp.co.soramitsu.fearless_utils.runtime.AccountId
+
+class RealProxySyncService(
+    private val chainRegistry: ChainRegistry,
+    private val proxyRepository: ProxyRepository,
+    private val accounRepository: AccountRepository,
+    private val accountDao: MetaAccountDao,
+    private val identityProvider: IdentityProvider
+) : ProxySyncService {
+
+    override suspend fun startSyncing() {
+        if (!accounRepository.hasMetaAccounts()) return
+
+        val metaAccounts = getMetaAccounts()
+
+        val supportedProxyChains = getSupportedProxyChains()
+        val chainsToAccountIds = supportedProxyChains.associateWith { chain -> chain.getAvailableAccountIds(metaAccounts) }
+
+        // proxiedsWithProxies will be usefull when we union differen proxy types to one account
+        val proxiedsWithProxies = chainsToAccountIds.flatMap { (chain, accountIds) ->
+            proxyRepository.getProxyDelegatorsForAccounts(chain.id, accountIds)
+        }
+
+        val newProxies = proxiedsWithProxies.formatToLocalProxies()
+        val oldProxies = accountDao.getAllProxyAccounts()
+
+        val proxiesDiff = CollectionDiffer.findDiff(newProxies, oldProxies, forceUseNewItems = false)
+
+        insertMetaAndChainAccounts(proxiesDiff)
+        insertProxies(proxiesDiff)
+    }
+
+    private suspend fun insertMetaAndChainAccounts(proxiesDiff: CollectionDiffer.Diff<ProxyAccountLocal>) {
+        val identitiesByChain = proxiesDiff.added.loadProxiedIdentities()
+
+        val chainAccounts = proxiesDiff.added.map { proxy ->
+            val identity = identitiesByChain[proxy.chainId]?.get(proxy.proxiedAccountId.intoKey())
+            val proxiedMetaId = accountDao.insertMetaAccountWithNewPosition { nextPosition ->
+                createMetaAccount(proxy.chainId, proxy.proxiedAccountId, identity, nextPosition)
+            }
+            createChainAccount(proxiedMetaId, proxy.chainId, proxy.proxiedAccountId)
+        }
+
+        accountDao.insertChainAccounts(chainAccounts)
+    }
+
+    private suspend fun insertProxies(proxiesDiff: CollectionDiffer.Diff<ProxyAccountLocal>) {
+        val deactivatedProxies = proxiesDiff.removed.map { it.copy(status = ProxyAccountLocal.Status.DEACTIVATED) }
```

Comment:
What is the reason we are not removing them from the database? To avoid wallet deletion somewhere in the middle of operation flow?

Reply:
No, I use status to show deactivated meta accounts in delegated updates screen.
But it's more relevant for MataAccountLocal so in another PR I moved Status class in MetaAccountLocal